mlir/lib/Dialect/Rock/Generator/Conv2dGenerator.cpp

```diff
@@ -815,7 +816,7 @@ LogicalResult Conv2dGenerator::genConvModule(ModuleOp &module, int kernel_id,
 
   std::vector<NamedAttribute> attributes{
       builder.getNamedAttr("gemm_id", builder.getI32IntegerAttr(gemmId)),
```

Comment:
Can `gemm_id` be made a proper attribute, with a comment in the `td` file,  that explains what it does? I am not sure I fully understand what is used for

Reply:
... good point, and, yes, I can go ahead and do that refactoring now.

gemm_id had two functions
- Marking which of the N different `rock.conv2d_bwd_data` operations a particular operation was so as to allow the backwards data algorithm (which is split into multiple kernels) to work correctly
- To mark which of several operations you were in in bwd weight and bwd data kernels so that we'd know which ones were utility kernels

Since this PR removes the hacky part of gemm_id, we can make the useful part (which bwd data op you're in) into an explicit attribute.

Good catch on the refactoring! I'll get on that

Reply:
Thanks for the explanation @krzysz00 ! Now it is everythign clearer :) !